JS-Classes/inheritance-es5.js

```diff
@@ -0,0 +1,85 @@
+function Car(name, model, year, color, maxSpeed, fuelCapacity = 60, fuelConsumption = 10) {
+    this.name = name;
+    this.model = model;
+    this.year = year;
+    this.color = color;
+    this.maxSpeed = maxSpeed;
+    this.fuelCapacity = fuelCapacity;
+    this.fuelConsumption = fuelConsumption;
+  }
+
+  Car.prototype.getFullName = function () {
+    return this.name + ' ' + this.model;
+  }
+
+  Car.prototype.getAge = function () {
+    return new Date().getFullYear() - this.year;
+  }
+
+  Car.prototype.changeColor = function (color) {
+    if (color == this.color) {
+      return `The car is already painted ${color}`;
+    } else {
+      this.color = color;
+      return `Great choice! Now your car has a ${color} color`;
+    }
+  }
+
+  Car.prototype.calculateWay = function (kilometers, fuel) {
+    const wayTime = (kilometers / this.maxSpeed).toFixed(1);
+    const fuelNeeded = (kilometers - fuel * 100 / this.fuelConsumption) * this.fuelConsumption / 100;
+    if (fuel < 10) {
+      return 'Fuel level less than 10L';
+    };
+    if (kilometers < 100 * (fuel / this.fuelConsumption)) {
+      return `The travel time will be ${wayTime}h. Refueling is not required`;
+    } else {
+      return `The travel time will be ${wayTime}h. ${fuelNeeded}L refueling is required`;
+    }
+  }
+
+  function CarWithOptions(name, model, year, color, maxSpeed, fuelCapacity = 60, fuelConsumption = 10, ...options) {
+    Car.call(this, name, model, year, color, maxSpeed, fuelCapacity = 60, fuelConsumption = 10);
```

Comment:
1. не нужно второй раз присваивать дефолтные значения аргументов, они присваиваются в родительском классе
```suggestion
  function CarWithOptions(name, model, year, color, maxSpeed, fuelCapacity, fuelConsumption, ...options) {
    Car.call(this, name, model, year, color, maxSpeed, fuelCapacity, fuelConsumption);
```
2. здесь (и в остальных классах) можно было сделать чуть по другому, чтобы меньше кода было.
просто поставить доп параметры в начало, а параметры базового класса в конец (но тогда options должен быть сразу массивом):
```suggestion
  function CarWithOptions(options, ...basicOptions) {
    Car.apply(this, basicOptions);
```